packages/legend-application-query/src/stores/QueryEditorStore.ts

```diff
@@ -157,16 +157,21 @@ export class QueryExportState {
   decorator?: ((query: Query) => void) | undefined;
   queryBuilderState: QueryBuilderState;
   persistQueryState = ActionState.create();
+  hideExportState?: boolean;
```

Comment:
the fact that we hack this thing so much makes me feel like the abstraction is too restrictive, would it be better if for `QueryEditorStore`, we have `saveState`, `saveAsState` and `renameState` as separate things? and generate 3 different modals for them, well arguable, `rename` can be a small popover or something.

The code looks like spaghetti bolognese now.